src/matches.rs

```diff
@@ -121,6 +121,30 @@ pub(crate) fn rewrite_match(
         inner_attrs[inner_attrs.len() - 1].span.hi()
     };
 
+    let blank_line_after_inner_attrs = if context.config.version() == Version::Two
+        && !inner_attrs.is_empty()
+        && !arms.is_empty()
+    {
+        let blank_line_snip = if let Some(hi) = context
+            .snippet(mk_sp(open_brace_pos, arms[0].span.lo()))
+            .find('/')
+        {
+            context.snippet(mk_sp(
+                open_brace_pos,
+                open_brace_pos + BytePos::from_usize(hi),
+            ))
+        } else {
+            context.snippet(mk_sp(open_brace_pos, arms[0].span.lo()))
+        };
+        if blank_line_snip.find("\n") != blank_line_snip.rfind("\n") {
+            "\n"
+        } else {
+            ""
+        }
+    } else {
+        ""
+    };
+
```

Comment:
It would be nicer to extract the `blank_line_after_inner_attrs` calculation into a standalone function so that `rewrite_match` isn't responsible for this logic.